featuretools/feature_base/feature_base.py

```diff
@@ -342,7 +342,9 @@ def from_dictionary(cls, arguments, entityset, dependencies, primitives_deserial
         entity_id = arguments['entity_id']
         variable_id = arguments['variable_id']
         variable = entityset[entity_id][variable_id]
-        return cls(variable)
+        cls_variable = cls(variable)
+        cls_variable._name = arguments['name']
```

Comment:
Since `_name` is intended to be private, I wonder if there is a better way to accomplish this. We could use `rename`, since it is the public API, but that would unnecessarily create a copy of the feature. Would it be worth adding an optional `name` keyword argument to the constructor?

Reply:
I added an optional `name` keyword argument to the constructor.